bson/src/main/org/bson/json/DateTimeFormatter.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.bson.json;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.time.Instant;
```

Comment:
No, it's ok.  Imports are compile-time only.  The static inner class that relies on it ends up in its own .class file at run time, and the rest of the code here protects us against it being used for Java 7 and lower.  You can see in https://evergreen.mongodb.com/version/590baedb2fbabe56ac003a25 that tests pass on all supported Java versions.